### PR TITLE
Add initial CI workflow

### PR DESCRIPTION
**Pull request by ulrikstrid**
Please review and fix anything that doesn't look right.

Using github actionns matrix feature: <https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix>

Base setup for nix: <https://nix.dev/tutorials/continuous-integration-github-actions.html>

nix-build-uncached: <https://github.com/Mic92/nix-build-uncached>

### CODE REVIEW

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+          nix_path: nixpkgs=channel:nixpkgs-unstable
+      - uses: cachix/cachix-action@v8
+        with:
+          name: mycache
+          # If you chose signing key for write access
+          signingKey: '${{ secrets.CACHIX_SIGNING_KEY }}'
+          # If you chose API tokens for write access OR if you have a private cache
+          authToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
+      - run: nix-env -i nix-build-uncached
+      - run: nix-build-uncached -build-flags "--experimental-features nix-command" ./sources.nix -A pkgs.ocaml-nng-ocamlPackages_${{ matrix.ocaml_version }}
```
Comment (anmonteiro) [inline .github/workflows/pull-request.yml:25]:
```suggestion
      - run: nix-build-uncached -build-flags "--experimental-features nix-command" ./sources.nix -A pkgs.ocaml-ng.ocamlPackages_${{ matrix.ocaml_version }}
```

File: .github/workflows/pull-request.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      matrix:
+        ocaml_version: [4_11, 4_12]
+        os: ["ubuntu-latest", "macos-latest"]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout
+      - uses: cachix/install-nix-action@v12
+      - uses: cachix/cachix-action@v8
+        with:
+          name: mycache
```
Comment (ulrikstrid) [inline .github/workflows/pull-request.yml:17]: @anmonteiro what should this line be?

File: .github/workflows/pull-request.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+          # If you chose signing key for write access
+          signingKey: '${{ secrets.CACHIX_SIGNING_KEY }}'
```
Comment (ulrikstrid) [inline .github/workflows/pull-request.yml:19]: Will you fix keys?

File: .github/workflows/pull-request.yml (hunk 2 above)
Comment (anmonteiro) [reply .github/workflows/pull-request.yml:19]: probably `anmonteiro`

Comment (anmonteiro) [reply .github/workflows/pull-request.yml:19]:
```suggestion
          name: anmonteiro
```

File: .github/workflows/pull-request.yml (hunk 3 above)
Comment (anmonteiro) [reply .github/workflows/pull-request.yml:21]: will only need the auth token

```suggestion
```

File: .github/workflows/pull-request.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+          # If you chose API tokens for write access OR if you have a private cache
+          authToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
+      - name: Build nix packages
+        run: |
+          nix-build ./sources.nix -A pkgs.ocaml-ng.ocamlPackages_${{ matrix.ocaml_version }}.piaf
+          nix-build ./sources.nix -A pkgs.ocaml-ng.ocamlPackages_${{ matrix.ocaml_version }}.oidc
+          nix-build ./sources.nix -A pkgs.ocaml-ng.ocamlPackages_${{ matrix.ocaml_version }}.tyxml-jsx
```
Comment (anmonteiro) [inline .github/workflows/pull-request.yml:28]:
```suggestion
          nix-build ./sources.nix -A pkgs.ocaml-ng.ocamlPackages_${{ matrix.ocaml_version }}.piaf \
          -A pkgs.ocaml-ng.ocamlPackages_${{ matrix.ocaml_version }}.oidc \
          -A pkgs.ocaml-ng.ocamlPackages_${{ matrix.ocaml_version }}.tyxml-jsx
```

**Comment by anmonteiro** [comment]
ok let's try this

**Pull request merged**